### PR TITLE
Additional F1 Web search options

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Per users request

### CODE REVIEW

File: src/Package/Impl/Help/SearchWebForCurrentCommand.cs [hunk 1]
```diff
@@ -32,11 +36,26 @@ internal sealed class SearchWebForCurrentCommand : HelpOnCurrentCommandBase {
...
             // Bing: search?q=item+site%3Astackoverflow.com
-            var encoded = HttpUtility.HtmlEncode(item);
-            var search = "http://" + Invariant($"www.bing.com/search?q={encoded}+R+site%3A{RToolsSettings.Current.HelpSearchSite}");
-            var browser = VsAppShell.Current.GetGlobalService<IVsWebBrowsingService>(typeof(SVsWebBrowsingService));
-            IVsWindowFrame frame;
-            browser.Navigate(search, (uint)__VSWBNAVIGATEFLAGS.VSNWB_WebURLOnly, out frame);
+            var tokens = RToolsSettings.Current.WebHelpSearchString.Split(new char[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
+            bool siteSet = false;
+
+            var sb = new StringBuilder("http://" + Invariant($"www.bing.com/search?q={HttpUtility.HtmlEncode(item)}"));
```
Comment (int19h) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: This should be `UrlEncode`. `HtmlEncode` will do things like `>` to `&gt;`, which will not be correct for an URI.

Comment (int19h) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: Better yet, `Uri.EscapeUriString`.

File: src/Package/Impl/Help/SearchWebForCurrentCommand.cs [hunk 2]
```diff
@@ -32,11 +36,26 @@ internal sealed class SearchWebForCurrentCommand : HelpOnCurrentCommandBase {
...
+            foreach (var t in tokens) {
+                sb.Append('+');
+                if (!siteSet && t.IndexOf('.') > 0) {
+                    sb.Append("site:");
+                    siteSet = true;
+                }
+                sb.Append(t);
```
Comment (int19h) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: `t` will need to be escaped here, too.

File: src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs [hunk 3]
```diff
@@ -104,7 +104,8 @@ internal sealed class RToolsSettingsImplementation : IRToolsSettings {
         public string SurveyNewsFeedUrl { get; set; } = SurveyNewsUrls.Feed;
         public string SurveyNewsIndexUrl { get; set; } = SurveyNewsUrls.Index;
         public bool EvaluateActiveBindings { get; set; } = true;
-        public string HelpSearchSite { get; set; } = "stackoverflow.com";
+        public string WebHelpSearchString { get; set; } = "R stackoverflow.com";
```
Comment (int19h) [inline src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs]: Should it be `R site:stackoverflow.com`?

File: src/Package/Impl/Help/SearchWebForCurrentCommand.cs [hunk 4]
```diff
@@ -32,11 +36,26 @@ internal sealed class SearchWebForCurrentCommand : HelpOnCurrentCommandBase {
...
-            IVsWindowFrame frame;
-            browser.Navigate(search, (uint)__VSWBNAVIGATEFLAGS.VSNWB_WebURLOnly, out frame);
+            var tokens = RToolsSettings.Current.WebHelpSearchString.Split(new char[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
+            bool siteSet = false;
+
+            var sb = new StringBuilder("http://" + Invariant($"www.bing.com/search?q={HttpUtility.HtmlEncode(item)}"));
+            foreach (var t in tokens) {
+                sb.Append('+');
+                if (!siteSet && t.IndexOf('.') > 0) {
+                    sb.Append("site:");
```
Comment (int19h) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: I'd rather let them specify it explicitly if they want site: or not. Not everything that contains a dot is necessarily a site.

Comment (int19h) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: (As in, just use the tokens as is, without trying to guess what's what. If they want site:foo.bar, they can put it there)

File: src/Package/Impl/Help/SearchWebForCurrentCommand.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: Yeah I thought about this but was not sure... OK then

File: src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs (hunk 3 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs]: No, token with dot auto-become site: parameter

File: src/Package/Impl/Help/SearchWebForCurrentCommand.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Help/SearchWebForCurrentCommand.cs]: OK, makes it simpler

**Pull request merged**